sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EChangeFeedITest.scala

```diff
@@ -272,6 +279,234 @@ class SparkE2EChangeFeedITest
+
+    for (sequenceNumber <- 1 to 50) {
+      val objectNode = Utils.getSimpleObjectMapper.createObjectNode()
+      objectNode.put("name", "Shrodigner's cat")
+      objectNode.put("type", "cat")
+      objectNode.put("age", 20)
+      objectNode.put("sequenceNumber", sequenceNumber)
+      objectNode.put("id", UUID.randomUUID().toString)
+      container.createItem(objectNode).block()
+    }
+
+    val checkpointLocation = "/checkpoints/" + UUID.randomUUID().toString
+    val cfg = Map(
+      "spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+      "spark.cosmos.accountKey" -> cosmosMasterKey,
+      "spark.cosmos.database" -> cosmosDatabase,
+      "spark.cosmos.container" -> cosmosContainer,
+      "spark.cosmos.changeFeed.itemCountPerTriggerHint" -> "1",
+      "spark.cosmos.read.inferSchema.enabled" -> "false",
+      "spark.cosmos.changeFeed.startFrom" -> "Beginning",
+      "spark.cosmos.read.partitioning.strategy" -> "Restrictive",
+      "spark.cosmos.changeFeed.batchCheckpointLocation" -> checkpointLocation
+    )
+
+    val df1 = spark.read.format("cosmos.oltp.changeFeed").options(cfg).load()
+    val rowsArray1 = df1.collect()
+    // technically possible that even with 50 documents randomly distributed across 3 partitions some
+    // has no documents
+    // rowsArray should have size df.rdd.getNumPartitions
+    rowsArray1.length > 0 shouldEqual true
+    rowsArray1.length <= df1.rdd.getNumPartitions shouldEqual true
+
+    val initialCount = rowsArray1.length
+
+    df1.schema.equals(
+      ChangeFeedTable.defaultIncrementalChangeFeedSchemaForInferenceDisabled) shouldEqual true
+
+    val hdfs = org.apache.hadoop.fs.FileSystem.get(spark.sparkContext.hadoopConfiguration)
+
+    val startOffsetFolderLocation = Paths.get(checkpointLocation, "startOffset").toString
+    val startOffsetFileLocation = Paths.get(startOffsetFolderLocation, "0").toString
+    hdfs.exists(new Path(startOffsetFolderLocation)) shouldEqual true
+    hdfs.exists(new Path(startOffsetFileLocation)) shouldEqual false
+
+    val latestOffsetFolderLocation = Paths.get(checkpointLocation, "latestOffset").toString
+    val latestOffsetFileLocation = Paths.get(latestOffsetFolderLocation, "0").toString
+    hdfs.exists(new Path(latestOffsetFolderLocation)) shouldEqual true
+    hdfs.exists(new Path(latestOffsetFileLocation)) shouldEqual true
+
+    hdfs.copyToLocalFile(true, new Path(latestOffsetFileLocation), new Path(startOffsetFileLocation))
+
+    val cfgWithoutItemCountPerTriggerHint = cfg.filter(keyValuePair => !keyValuePair._1.equals("spark.cosmos.changeFeed.itemCountPerTriggerHint"))
+    val df2 = spark.read.format("cosmos.oltp.changeFeed").options(cfgWithoutItemCountPerTriggerHint).load()
+    val rowsArray2 = df2.collect()
+    rowsArray2 should have size 50 - initialCount
+  }
+
+  "spark change feed query (incremental)" can "proceed with simulated Spark2 Checkpoint" in {
+    val cosmosEndpoint = TestConfigurations.HOST
```

Comment:
Not related to this PR. But this test is failing for me because of:
1. java.io.IOException: mkdir of /checkpoints/546ac653-0bd7-4469-87ef-346696f4fb3b/startOffset failed. -> do I need to first give write permission to the folder? 
2. `The retention duration in Change Feed policy can only be enabled when Full-Fidelity Change Feed is enabled for database account` -> 
```
    properties.setChangeFeedPolicy(
      ChangeFeedPolicy.createFullFidelityPolicy(Duration.ofMinutes(10)))
```
 how do we guarantee that the database account created during the pipeline is full-fidelity feature opt-int?

Reply:
Permission: not sure might be difference between iOS and Windows. For me it wokred without any explicit permission granting.

Reply:
ah so this test Is not part of the pipeline

Reply:
Which test? "spark change feed query (incremental)" can "proceed with simulated Spark2 Checkpoint" in { is runnign int he pipeline
But it runs on WIndows